src/us/kbase/test/auth2/lib/AuthenticationAnonymousIDsTest.java

```diff
@@ -0,0 +1,193 @@
+		final UUID uid3 = UUID.randomUUID();
+		mockret.put(UID1, new UserName("foo"));
+		mockret.put(UID2, new UserName("bar"));
+		mockret.put(uid3, new UserName("***ROOT***"));
+		when(storage.getUserNamesFromAnonymousIDs(set(UID2, UID1, uid3))).thenReturn(mockret);
+		
+		final Map<UUID, UserName> ret = auth.getUserNamesFromAnonymousIDs(token, set(
+				UID1, UID2, uid3));
+		
+		assertThat("incorrect user map", ret, is(expected));
+	}
+	
+	@Test
+	public void getUserNamesFromAnonymousIDsFailNulls() throws Exception {
+		final Authentication auth = initTestMocks().auth;
+		
+		final IncomingToken t = new IncomingToken("foobar");
+		final Set<UUID> u = set(UID1);
+		
+		getUserNamesFromAnonymousIDsFail(auth, null, u, new NullPointerException("token"));
+		getUserNamesFromAnonymousIDsFail(auth, t, null, new NullPointerException("anonymousIDs"));
+		getUserNamesFromAnonymousIDsFail(auth, t, set(UID1, null), new NullPointerException(
+				"Null ID in anonymousIDs"));
+	}
+	
+	@Test
+	public void getUserNamesFromAnonymousIDsExecuteStandardUserCheckingTests() throws Exception {
+		final IncomingToken token = new IncomingToken("foo");
+		AuthenticationTester.executeStandardUserCheckingTests(
+				new AbstractAuthOperation() {
+			
+					@Override
+					public IncomingToken getIncomingToken() {
+						return token;
+					}
+					
+					@Override
+					public void execute(final Authentication auth) throws Exception {
+						auth.getUserNamesFromAnonymousIDs(token, set(UID1));
+					}
+		
+					@Override
+					public List<ILoggingEvent> getLogAccumulator() {
+						return logEvents;
+					}
+					
+					@Override
+					public String getOperationString() {
+						return "translate anonymous IDs";
+					}
+				},
+				set(Role.DEV_TOKEN, Role.SERV_TOKEN, Role.CREATE_ADMIN, Role.ROOT),
+				set()
+		);
+	}
+	
+	@Test
+	public void getUserNameFromAnonymousIDsFailTooManyInputs() throws Exception {
+		// includes test that serv tokens work
+		final TestMocks testauth = initTestMocks();
+		final AuthStorage storage = testauth.storageMock;
+		final Authentication auth = testauth.auth;
+		
+		final IncomingToken token = new IncomingToken("foobar");
+		
+		final Set<UUID> inputs = Stream.generate(UUID::randomUUID)
+				.limit(10001).collect(Collectors.toSet());
+		
+		when(storage.getToken(token.getHashedToken())).thenReturn(
+				StoredToken.getBuilder(TokenType.SERV, UUID.randomUUID(), new UserName("admin"))
+						.withLifeTime(Instant.now(), Instant.now()).build());
+		when(storage.getUser(new UserName("admin"))).thenReturn(AuthUser.getBuilder(
+				new UserName("admin"), UUID.randomUUID(), new DisplayName("d"), NOW)
+				.withRole(Role.ADMIN)
+				.build());
+		
+		getUserNamesFromAnonymousIDsFail(auth, token, inputs, new IllegalParameterException(
+				"Anonymous ID count exceeds maximum of 10000"));
+	}
+	
+	private void getUserNamesFromAnonymousIDsFail(
+			final Authentication auth,
+			final IncomingToken token,
+			final Set<UUID> anonymoutIDs,
```

Comment:
Typo: `anonymoutIDs`

Reply:
I'll fix this in a later PR if you don't mind. Fixed it locally

Reply:
No prob, not a biggie, just a thing I noticed.